staging/src/k8s.io/apiserver/pkg/apis/apiserver/types.go

```diff
@@ -48,3 +48,47 @@ type AdmissionPluginConfiguration struct {
 	// +optional
 	Configuration *runtime.Unknown
 }
+
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+
+// ConnectivityServiceConfiguration provides versioned configuration for network proxy clients.
+type ConnectivityServiceConfiguration struct {
+	metav1.TypeMeta
+
+	// ConnectionServices contains a list of network proxy client configurations
+	ConnectionServices []ConnectionService
+}
+
+// ConnectionService provides the configuration for a single network proxy client.
+type ConnectionService struct {
+	// Name is the name of the connectivity service.
+	// Currently supported values are "Master" and "Cluster"
```

Comment:
It's been a while, but I seem to recall talking about etcd as well.

Reply:
We will have etcd as well. However I have not added that support in this PR. Was planning on fixing that then.